Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CentralRepoPostgresSettingsUtil.java

```diff
@@ -0,0 +1,177 @@
+import org.sleuthkit.datamodel.CaseDbConnectionInfo;
+
+/**
+ * handles saving and loading of postgres settings for central repo
+ */
+public class CentralRepoPostgresSettingsUtil {
+    private final static Logger LOGGER = Logger.getLogger(CentralRepoPostgresSettingsUtil.class.getName());
+    
+    private final static String DEFAULT_HOST = ""; // NON-NLS
+    private final static int DEFAULT_PORT = 5432;
+    private final static String DEFAULT_DBNAME = "central_repository"; // NON-NLS
+    private final static String DEFAULT_USERNAME = "";
+    private final static String DEFAULT_PASSWORD = "";
+
+    private static final String PASSWORD_KEY = "db.postgresql.password";
+    private static final String BULK_THRESHOLD_KEY = "db.postgresql.bulkThreshold";
+    private static final String PORT_KEY = "db.postgresql.port";
+    private static final String USER_KEY = "db.postgresql.user";
+    private static final String DBNAME_KEY = "db.postgresql.dbName";
+    private static final String HOST_KEY = "db.postgresql.host";
+
+    private static final String MODULE_KEY = "CentralRepository";
+
+    
+    
+    private static String valOrDefault(String val, String defaultVal) {
+        if (val == null || val.isEmpty())
+            return defaultVal;
+        
+        return val;
+    }
+    
+    private static int valOrDefault(String val, int defaultVal, Integer min, Integer max) {
+        try {
+            if (val == null || val.isEmpty()) {
+                return defaultVal;
+            } else {
+                int retVal = Integer.parseInt(val);
+                if ((min != null && retVal < min) || (max != null && retVal > max)) {
+                    return defaultVal;
+                }
+                else {
+                    return retVal;
+                }
+            }
+        } catch (NumberFormatException ex) {
+            return defaultVal;
+        }
+    }
+    
+    private static void handleTry(TryHandler handler) {
+        try {
+            handler.operation();
+        }
+        catch (CentralRepoException e) {
+            LOGGER.log(Level.WARNING, "There was an error in converting central repo postgres settings", e);
+        }
+    }
+    
+    /**
+     * an action that potentially throws an exception
+     */
+    private interface TryHandler {
+        void operation() throws CentralRepoException;
+    }
+    
+    
+    public static PostgresConnectionSettings loadMultiUserSettings() {
+        PostgresConnectionSettings settings = new PostgresConnectionSettings();
+        
+        CaseDbConnectionInfo muConn;
+        try {
+            muConn = UserPreferences.getDatabaseConnectionInfo();
+        } catch (UserPreferencesException ex) {
+            LOGGER.log(Level.SEVERE, "Failed to import settings from multi-user settings.", ex);
+            return settings;
+        }
+        
+        handleTry(() -> settings.setHost(valOrDefault(muConn.getHost(), DEFAULT_HOST)));
```

Comment:
Should we do more than just log an error if these parameters are invalid? There may be empty fields in the settings object. If this is what we want to do, add a comment that this is what's happening.

Reply:
My goal for this and the other thing you mentioned in this file was to preserve the the same behavior here: https://github.com/sleuthkit/autopsy/blob/develop/Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/PostgresCentralRepoSettings.java#L73.  Would it be worth changing right now or should this wait for another time?

Reply:
I don't think you've quite preserved that logic, though I may be reading it wrong. It seems like in your new code if the exception gets thrown the field doesn't get saved to the default.

Reply:
Oh okay.  I'll fix that.